tests/cpp/CMakeLists.txt

```diff
@@ -32,6 +32,11 @@ set(CMAKE_VERBOSE_MAKEFILE ON)
 set(CMAKE_BUILD_TYPE DEBUG)
 set(CMAKE_CXX_STANDARD 17)
 
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_C_COMPILER_ID STREQUAL "GNU"))
+    set(CMAKE_CXX_FLAGS --coverage)
+    add_link_options(--coverage)
+endif()
+
```

Comment:
See previous comment

Reply:
Now this is contingent on -DCOVERAGE=on being passed to the cmake command

Reply:
Given that coverage is built into the library we're linking against, is this actually just doing coverage analysis of the tests themselves rather than the library code?

Reply:
Yes, but we don't capture the coverage for the tests and so they are not included in our overall coverage. However, the compile and link options need to be there for the code hits to be registered properly.